authors/apps/articles/views.py

```diff
@@ -420,3 +422,89 @@ def delete(self, request, *args, **kwargs):
         comment.soft_delete()
         return Response({"detail": "comment deleted"},
                         status=status.HTTP_200_OK)
+
+
+class BookmarkView(generics.CreateAPIView):
+    """This class creates a new like or dislike"""
+    permission_classes = (IsAuthenticated,)
+    serializer_class = BookmarkSerializer
+
+    def post(self, request, slug):
+        """Creates a bookmark"""
+        data = request.data
+        article = Article.objects.filter(
+            slug=slug, published=True, activated=True).first()
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+
+        bookmark = Bookmark.objects.filter(
+            user_id=request.user.pk, article_id=article.id).first()
+        if bookmark is not None:
+            favorite_found = {
+                "detail": "Article already bookmarked."
+            }
+            return Response(data=favorite_found,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        data['article_id'] = article.id
+        data['user_id'] = request.user.pk
+        serializer = self.get_serializer(data=data)
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+        data['detail'] = 'Article added to bookmarks.'
+        return Response(data, status=status.HTTP_201_CREATED)
+
+    def delete(self, request, *args, **kwargs):
+        article = Article.objects.filter(
+            slug=kwargs['slug'], published=True, activated=True).first()
+        if article is None:
+            not_found = {
+                "detail": "This article has not been found."
+            }
+            return Response(data=not_found, status=status.HTTP_404_NOT_FOUND)
+        favorite = Bookmark.objects.filter(
```

Comment:
Please consider using the `.get()` I believe it's more explicit.
try.
```suggestion
        Article.objects.filter(published=True, activated=True).get(slug=kwargs['slug'])
```

Reply:
This will break my code because `.get()` throws an error if nothing is found